ste/mgr-JobPartMgr.go

```diff
@@ -640,6 +642,9 @@ var builtinTypes = map[string]string{
 func (jpm *jobPartMgr) inferContentType(fullFilePath string, dataFileToXfer []byte) string {
 	fileExtension := filepath.Ext(fullFilePath)
 
+	if contentType, ok := environmentMimeMap[strings.ToLower(fileExtension)]; ok {
+		return contentType
+	}
```

Comment:
Ohh I see. 
1. Try to infer MIMEType from user specified file. (which we've added now)
2. Then try to infer from built-In-Types
3. Then try to infer from the mime.go lib provided by golang
4. And lastly from the first 512B (headers) of the file.